runtime/src/lib.rs

```diff
@@ -549,13 +543,75 @@ mod parachain_impl {
         }
     }
 
+    fn native_currency_location(id: ForeignCurrencyId) -> MultiLocation {
+        X3(
+            Parent,
+            Parachain {
+                id: ParachainInfo::get().into(),
+            },
+            GeneralKey(id.encode()),
+        )
+    }
+
+    pub struct CurrencyIdConvert;
```

Comment:
The only thing bothering me a bit is this block of the conversion logic, can we have it in a file next to the lib.rs?